### ISSUE TITLE
Implement `FromIterator` for `SharedString`

### ISSUE THREAD

**Issue by Enyium** [slint-ui/slint#6232]
I implemented a global-singleton callback in Rust that edits a `SharedString` by iterating over its characters, filtering them and `collect()`ing them back into a `String`, then using `.into()` to return a `SharedString` again.

Looking at the source code, this copies the data unnecessarily. If `SharedString` implemented `FromIterator`, I could directly `collect()` into a `SharedString`.

**Comment by ogoffart** [comment]
True, it could implement it just like String from std implements it.
